pkg/providers/instancetype/suite_test.go

```diff
@@ -1679,6 +1675,57 @@ var _ = Describe("InstanceType Provider", func() {
 			Expect(err.Error()).To(ContainSubstring("creating instance failed"))
 		})
 	})
+
+	Context("SKU Scaling", func() {
+		It("should scale up and down with a random SKU per family", func() {
+			// Get a few example SKUs from the working SKUs list
+			workingSKUs := instancetype.GetKarpenterWorkingSKUs()
+			Expect(workingSKUs).ToNot(BeEmpty())
+
+			// Test with a few specific SKUs
+			testSKUs := []string{"Standard_A2", "Standard_B1s"}
+
+			for _, skuName := range testSKUs {
+				fmt.Println("Current SKU:", skuName)
+				skuNodePool := coretest.NodePool(karpv1.NodePool{
+					Spec: karpv1.NodePoolSpec{
+						Template: karpv1.NodeClaimTemplate{
+							Spec: karpv1.NodeClaimTemplateSpec{
+								NodeClassRef: &karpv1.NodeClassReference{
+									Group: object.GVK(nodeClass).Group,
+									Kind:  object.GVK(nodeClass).Kind,
+									Name:  nodeClass.Name,
+								},
+								Requirements: []karpv1.NodeSelectorRequirementWithMinValues{
+									{
+										NodeSelectorRequirement: v1.NodeSelectorRequirement{
+											Key:      v1beta1.LabelSKUName,
+											Operator: v1.NodeSelectorOpIn,
+											Values:   []string{skuName},
+										},
+									},
+								},
+							},
+						},
+					},
+				})
```

Comment:
why create nodepools instead of creating a nodeclaim directly? 

Also should we be testing that these skus work with specific image families for both ubuntu + azure linux? Some GPUs are only supported for ubuntu for example?